platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -69,43 +65,73 @@
+  /**
+   * Cancels scheduled velocity animations if user doesn't lift fingers within
+   * {@link MapboxConstants#SCHEDULED_ANIMATION_TIMEOUT}
+   */
+  private Handler animationsTimeoutHandler = new Handler();
 
   MapGestureDetector(Context context, Transform transform, Projection projection, UiSettings uiSettings,
-                     TrackingSettings trackingSettings, AnnotationManager annotationManager,
-                     CameraChangeDispatcher cameraChangeDispatcher) {
+                     AnnotationManager annotationManager, CameraChangeDispatcher cameraChangeDispatcher) {
     this.annotationManager = annotationManager;
     this.transform = transform;
     this.projection = projection;
     this.uiSettings = uiSettings;
-    this.trackingSettings = trackingSettings;
     this.cameraChangeDispatcher = cameraChangeDispatcher;
 
-    // Touch gesture detectors
+    // Checking for context != null for testing purposes
     if (context != null) {
-      gestureDetector = new GestureDetectorCompat(context, new GestureListener());
-      gestureDetector.setIsLongpressEnabled(true);
-      scaleGestureDetector = new ScaleGestureDetector(context, new ScaleGestureListener());
-      ScaleGestureDetectorCompat.setQuickScaleEnabled(scaleGestureDetector, true);
-      rotateGestureDetector = new RotateGestureDetector(context, new RotateGestureListener());
-      shoveGestureDetector = new ShoveGestureDetector(context, new ShoveGestureListener());
+      gesturesManager = new AndroidGesturesManager(context);
+
+      Set<Integer> shoveScaleSet = new HashSet<>();
+      shoveScaleSet.add(AndroidGesturesManager.GESTURE_TYPE_SHOVE);
+      shoveScaleSet.add(AndroidGesturesManager.GESTURE_TYPE_SCALE);
+
+      Set<Integer> shoveRotateSet = new HashSet<>();
+      shoveRotateSet.add(AndroidGesturesManager.GESTURE_TYPE_SHOVE);
+      shoveRotateSet.add(AndroidGesturesManager.GESTURE_TYPE_ROTATE);
+
+      Set<Integer> ScaleLongPressSet = new HashSet<>();
+      ScaleLongPressSet.add(AndroidGesturesManager.GESTURE_TYPE_SCALE);
+      ScaleLongPressSet.add(AndroidGesturesManager.GESTURE_TYPE_LONG_PRESS);
+
+      gesturesManager.setMutuallyExclusiveGestures(shoveScaleSet, shoveRotateSet, ScaleLongPressSet);
+
+      gesturesManager.setStandardGestureListener(new StandardGestureListener());
+      gesturesManager.setMoveGestureListener(new MoveGestureListener());
+      gesturesManager.setStandardScaleGestureListener(new ScaleGestureListener(
+        context.getResources().getDimension(R.dimen.mapbox_minimum_scale_velocity)
+      ));
+      gesturesManager.setRotateGestureListener(new RotateGestureListener(
+        context.getResources().getDimension(R.dimen.mapbox_minimum_scale_span_when_rotating),
+        context.getResources().getDimension(R.dimen.mapbox_minimum_angular_velocity)
+      ));
+      gesturesManager.setShoveGestureListener(new ShoveGestureListener());
+      gesturesManager.setMultiFingerTapGestureListener(new TapGestureListener());
```

Comment:
not actionable for now, but for future iterations is to make the initialisation code of the gesture library more discoverable? What I mean with that is that user atm needs to copy paste the code while with something as a builder/factory they could discover the api through specific components and code completion.